### ISSUE TITLE
Python 3 support

### ISSUE THREAD

**Issue by wavexx** [wavexx/screenkey#3]
Both `python-xlib` and `python-xcb` are currently out-of-date for python 3. This blocks python 3 support in Screenkey.

**Comment by wavexx** [comment]
python-xlib is no longer used and Screenkey/keylistener.py works correctly with gi/GLib and python3.
pygtk is not available on python3 though, meaning that a full transition to gi/gtk3 is required.

**Comment by rhoit** [comment]
how can i test the python3 branch

**Comment by wavexx** [comment]
There's no branch for python 3. The internal key handling library is python3-compatible, but the current GUI is still based on gtk2 and must be rewritten for python-gi.

**Comment by rhoit** [comment]
thanx, I'm trying to port into py3, I have come up to bare workable [version](https://github.com/rhoit/screenkey) but much things to be done.

**Comment by wavexx** [comment]
Strictly speaking, I'd like to ditch most of the current gui building code and use something like glade to construct the UI instead. Several parts of the listener also need changes to display mouse activity, which is now recorded but didn't finish displaying the status yet.

**Comment by aleb** [comment]
How is this going? Need help?

**Comment by rhoit** [comment]
kinda on hold buzy in other things..

**Comment by aleb** [comment]
@rhoit Could you summarize what else is needed until it's complete? I might spend some time on it.
Is there interest in this - @wavexx at what point would you consider it mergeable?

**Comment by aleb** [comment]
@wavexx could you comment about https://github.com/rhoit/screenkey/commits/py3-port ? Do you consider it a good start for the port to Python3?

**Comment by wavexx** [comment]
On Tue, Nov 29 2016, Alex Băluț wrote:
> @rhoit Could you summarize what else is needed until it's complete? I might spend some time on it.
> Is there interest in this - @wavexx at what point would you consider it mergeable?

I'd actually like to get rid of the gi code and use a gui builder for
the preferences.

I also have locally uncommitted changes to support mouse clicks which
break most of those.

Do not change style unless needed guys!

**Comment by rhoit** [comment]
I haven't worked much in the preferences, the core thing is runnable, but it doesn't render the symbols. I will get back with screenshot and detail later till tommorow got to run now.

**Comment by wavexx** [comment]
On Tue, Nov 29 2016, Alex Băluț wrote:
> @wavexx could you comment about https://github.com/rhoit/screenkey/commits/py3-port ? Do you consider it a good start for the port to Python3?

I'll take a look in the next days, but I've been quite busy with other
projects lately.

I don't actually think the gtk3 port is a big issue, there isn't much
code involved. And most of the display code needs to be changed for
mouse clicks anyway, so there won't be much left.

The main issue is really just the prefs, and if we migrate to something
like glade it won't be an issue.

**Comment by rhoit** [comment]
Things to be done

- back space, arrow keys and space icon rendering
- build preference, make change preference to reflect in UI
- show mouse information.
- movable ui widget and auto font resizing.
- fix key modes

here are the screen shot for now and then

port version

![screenshot96](https://cloud.githubusercontent.com/assets/1684754/20761965/d2ddf2b6-b74c-11e6-95ac-7e6c085c49cc.png)

![screenshot95](https://cloud.githubusercontent.com/assets/1684754/20761731/0855d7ca-b74c-11e6-9334-eb7135f15d9b.png)
 
![screenshot92](https://cloud.githubusercontent.com/assets/1684754/20761616/a6786090-b74b-11e6-8dd0-f82ff9d8961c.png)

old version

![screenshot94](https://cloud.githubusercontent.com/assets/1684754/20761728/06179228-b74c-11e6-88e6-ebba7ff563dc.png)

![screenshot93](https://cloud.githubusercontent.com/assets/1684754/20761623/a8f8df2a-b74b-11e6-907d-8113ca9f5a34.png)
